opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java

```diff
@@ -0,0 +1,66 @@
+package io.opentracing.contrib.spring.web;
+
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureOrder;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.Ordered;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+import io.opentracing.NoopTracer;
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.web.interceptor.SpanDecorator;
+import io.opentracing.contrib.spring.web.interceptor.TracingHandlerInterceptor;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+
+/**
+ * @author Pavol Loffay
+ */
+@Configuration
+@AutoConfigureOrder(Ordered.HIGHEST_PRECEDENCE)
+@ConditionalOnMissingBean(TracingFilter.class)
+public class TracingAutoconfiguration {
+    private static final Logger log = Logger.getLogger(TracingAutoconfiguration.class.getName());
+
+    @Autowired
+    private Tracer tracer;
+
+    @Bean
+    @ConditionalOnMissingBean(Tracer.class)
+    public Tracer noopTracer() {
+        log.severe("Tracer bean is not configured! Switching to " + NoopTracer.class.getName());
+        return NoopTracerFactory.create();
+    }
+
+    @Bean
+    public FilterRegistrationBean tracingFilter() {
+        TracingFilter tracingFilter = new TracingFilter(tracer,
+                Arrays.asList(io.opentracing.contrib.web.servlet.filter.SpanDecorator.STANDARD_TAGS));
+
+        FilterRegistrationBean filterRegistrationBean = new FilterRegistrationBean(tracingFilter);
+        filterRegistrationBean.addUrlPatterns("/*");
+        filterRegistrationBean.setOrder(Integer.MIN_VALUE);
+        filterRegistrationBean.setAsyncSupported(true);
+
+        return filterRegistrationBean;
+    }
+
+    @Bean
+    public WebMvcConfigurerAdapter tracingHandlerInterceptor() {
+        return new WebMvcConfigurerAdapter() {
+            @Override
+            public void addInterceptors(InterceptorRegistry registry) {
+                registry.addInterceptor(new TracingHandlerInterceptor(tracer,
+                        Arrays.asList(SpanDecorator.STANDARD_TAGS)));
```

Comment:
Yes, I will think of a better name in following PR. IRRC interceptor's `STANDARD_TAGS` decorator adds only logs..